modules/fledge.js

```diff
@@ -0,0 +1,192 @@
+  const gptSlot = getGptSlotForAdUnitCode(adUnitCode);
+  logWarn(MODULE, 'gptSlotInfo', gptSlot);
+  if (gptSlot && gptSlot.setConfig) {
+    delete componentAuctionConfig.bidId;
+    gptSlot.setConfig({
+      componentAuction: [{
+        configKey: seller,
+        auctionConfig: componentAuctionConfig
+      }]
+    });
+    logInfo(MODULE, `register component auction config in GPT for: ${adUnitCode} x ${seller}: ${gptSlot.getAdUnitPath()}`, componentAuctionConfig);
+  } else {
+    logWarn(MODULE, `unable to register component auction config in GPT for: ${adUnitCode} x ${seller}.`);
+  }
+
+  next(bidRequest, componentAuctionConfig);
+}
+
+export function addComponentAuctionHook(next, bidRequest, fledgeAuctionConfig) {
+  if (!bidRequest || !fledgeAuctionConfig) { return; }
+
+  const seller = fledgeAuctionConfig.seller;
+  const adUnitCode = bidRequest.adUnitCode;
+
+  // TODO : remove duplicate {adUnitCode, seller} is already registered
+  logInfo(MODULE, `adding auction config for ${adUnitCode} x ${seller}`);
+  _componentAuctions.push({adUnitCode, seller, fledgeAuctionConfig});
+
+  next(bidRequest, fledgeAuctionConfig);
+}
+
+export function getAuctionConfig(bid) {
+  const targetAdUnitCode = bid.adUnitCode;
+  const componentAuctionsForAdUnitCode = _componentAuctions.filter((componentAuction) => {
+    const {adUnitCode} = componentAuction;
+    return targetAdUnitCode == adUnitCode;
+  });
+  if (componentAuctionsForAdUnitCode.length == 0) {
+    logInfo(MODULE, `NOT running runAdAuction for adUnitCode=${targetAdUnitCode} since there are no componentAuctions`);
+    return null;
+  }
+
+  logInfo(MODULE, `runAdAuction for adUnitCode=${targetAdUnitCode}`, componentAuctionsForAdUnitCode);
+
+  const seller = config.getConfig('fledge.seller');
+  const decisionLogicUrl = config.getConfig('fledge.decisionLogicUrl');
+  if (seller && decisionLogicUrl) {
+    // multiple-ssp with top level support
+    return {
+      seller,
+      decisionLogicUrl,
+      componentAuctions: componentAuctionsForAdUnitCode.map((auction) => auction.fledgeAuctionConfig)
+    }
+  }
+
+  // fallback if there is no top-level seller defined
+  const randomIdx = Math.floor(Math.random() * componentAuctionsForAdUnitCode.length);
+  return componentAuctionsForAdUnitCode[randomIdx].fledgeAuctionConfig;
+}
+
+export function renderAdHook(next, doc, id, options) {
+  logInfo(MODULE, 'renderAd hook', arguments);
+  if (doc && id) {
+    try {
+      const bid = auctionManager.findBidByAdId(id);
+
+      if (bid) {
+        logInfo(MODULE, 'renderAd hook found bid', bid);
+        if (runAdAuction(bid, next, doc, id, options)) {
+          return;
+        }
+      }
+    } catch (e) {
+      logError(MODULE, 'fail running fledge', id, e);
+    }
+  }
+  next(doc, id, options);
+}
+
+export function runAdAuction(bid, next, doc, id, options) {
+  if (bid.mediaType == 'video') {
+    logWarn(MODULE, 'Video fledge ads not yet supported.');
```

Comment:
I don't think native would work either, since it can't be reduced to a single URL.